### PR TITLE
API Services for Snapshot fetch & Submit question

### PR DESCRIPTION
**Pull request by rajeevbbqq**

### CODE REVIEW

File: infrastructure/repository/snapshot_repository.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from infrastructure.models import Snapshots, TransferInfo
+
+
+class Snapshot(BaseRepository):
+    def find_by_address(self, address):
+
+        result = (
+            self.session.query(Snapshots)
+            .filter(Snapshots.address == address)
+            .limit(1)
```
Comment (raamb) [inline infrastructure/repository/snapshot_repository.py:11]: why limit?

File: infrastructure/repository/snapshot_repository.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            .first()
+        )
+
+        return result or None
+
+    def find_transfer_status(self, address):
+        result = (
+            self.session.query(TransferInfo)
+            .filter(TransferInfo.wallet_address == address)
+            .limit(1)
```
[the remark this answers is not in the document]
Comment (rajeevbbqq) [reply infrastructure/repository/snapshot_repository.py]: Changed to first() and removed limit

File: infrastructure/repository/snapshot_repository.py (hunk 1 above)
Comment (rajeevbbqq) [reply infrastructure/repository/snapshot_repository.py]: Changed to first() and removed limit

File: infrastructure/repository/token_snapshot_repo.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+from infrastructure.models import Snapshots, TransferInfo
+from domain.factory.token_snapshot_factory import TokenSnapshotFactory
+from sqlalchemy import exc
+
+
+class TokenSnapshotRepo(BaseRepository):
+    def get_token_balance(self, address):
+
+        try:
+            result = (
```
Comment (prashantramangupta) [inline infrastructure/repository/token_snapshot_repo.py:11]: DB Session is not handled properly.

File: infrastructure/repository/token_snapshot_repo.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            transfer_details = self.get_transfer_status(address)
+            return TokenSnapshotFactory.convert_token_snapshot_db_to_entity_model(
+                result, transfer_details
+            ).to_response()
+        else:
+            return None
+
+    def get_transfer_status(self, address):
+        try:
+            result = (
```
Comment (prashantramangupta) [inline infrastructure/repository/token_snapshot_repo.py:29]: Db Session is not handled properly

File: infrastructure/repository/customer_comments_repo.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+        try:
+            user_comment = UserComments(wallet_address=address, comment=comment)
+            user_access_log = UserAccessDetails(
+                wallet_address=address,
+                email=email,
+                name=name,
+                login_time=datetime.now(),
+            )
+
```
Comment (prashantramangupta) [inline infrastructure/repository/customer_comments_repo.py:18]: We don't need to add try and except here as add_item takes care of it.

File: infrastructure/repository/token_snapshot_repo.py (hunk 3 above)
Comment (rajeevbbqq) [reply infrastructure/repository/token_snapshot_repo.py:11]: Fix in #4

File: infrastructure/repository/token_snapshot_repo.py (hunk 4 above)
Comment (rajeevbbqq) [reply infrastructure/repository/token_snapshot_repo.py:29]: Fix in #4

File: infrastructure/repository/customer_comments_repo.py (hunk 5 above)
Comment (rajeevbbqq) [reply infrastructure/repository/customer_comments_repo.py:18]: Fix in #4

**Pull request merged**